[MDB Ignore] Modsuits. In 2029.

Tsar-Salat opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

closes https://github.com/BeeStation/BeeStation-Hornet/issues/12086

Title. This be modsuits, beware all ye who enter.

Modsuits are effectively a modular spacesuit base similiar to hardsuits, but with each clothing slot being retractable. Modsuits ship with 20+ different modules for science, medical, engineering, command, supply, and centcom.

Its intended in this PR to convert all Head positions and antagonists to use modsuits, with engineers and cargo getting the rest. 

Mining, Atmospherics, and Security retain their hardsuits at round start, but its possible to build their respective suits during the course of the round.



### My Code

My code consists of:
- lowered slowdowns compared to tg
- AIs cannot be taken out of modsuits once locked. They must unlock the modsuit themselves.
- pathfinder plus module
- Keeping hardsuits as an alternative for combat roles.


Ports:
- https://github.com/tgstation/tgstation/pull/55769
- https://github.com/tgstation/tgstation/pull/59109
- https://github.com/tgstation/tgstation/pull/79674 (shielded fix)



<details>
<summary>MODsuit additions (exhaustive, not really necessary to list them all, but to credit all authors)</summary>

- https://github.com/tgstation/tgstation/pull/61138
- https://github.com/tgstation/tgstation/pull/62907
- https://github.com/tgstation/tgstation/pull/63755
- https://github.com/tgstation/tgstation/pull/63638
- https://github.com/tgstation/tgstation/pull/63670
- https://github.com/tgstation/tgstation/pull/63706
- https://github.com/tgstation/tgstation/pull/63745
- https://github.com/tgstation/tgstation/pull/63749
- https://github.com/tgstation/tgstation/pull/63791
- https://github.com/tgstation/tgstation/pull/63795
- https://github.com/tgstation/tgstation/pull/63819
- https://github.com/tgstation/tgstation/pull/63836
- https://github.com/tgstation/tgstation/pull/63999
- https://github.com/tgstation/tgstation/pull/64042
- https://github.com/tgstation/tgstation/pull/64064
- https://github.com/tgstation/tgstation/pull/64151
- https://github.com/tgstation/tgstation/pull/64237
- https://github.com/tgstation/tgstation/pull/64359
- https://github.com/tgstation/tgstation/pull/64668
- https://github.com/tgstation/tgstation/pull/64674
- https://github.com/tgstation/tgstation/pull/64709
- https://github.com/tgstation/tgstation/pull/64857
- https://github.com/tgstation/tgstation/pull/64934
- https://github.com/tgstation/tgstation/pull/65000
- https://github.com/tgstation/tgstation/pull/65193
- https://github.com/tgstation/tgstation/pull/65416
- https://github.com/tgstation/tgstation/pull/65445
- https://github.com/tgstation/tgstation/pull/65559
- https://github.com/tgstation/tgstation/pull/65733
- https://github.com/tgstation/tgstation/pull/65753
- https://github.com/tgstation/tgstation/pull/65915
- https://github.com/tgstation/tgstation/pull/66087
- https://github.com/tgstation/tgstation/pull/66170
- https://github.com/tgstation/tgstation/pull/66190
- https://github.com/tgstation/tgstation/pull/66196
- https://github.com/tgstation/tgstation/pull/66208
- https://github.com/tgstation/tgstation/pull/66520
- https://github.com/tgstation/tgstation/pull/66574
- https://github.com/tgstation/tgstation/pull/66667
- https://github.com/tgstation/tgstation/pull/67042
- https://github.com/tgstation/tgstation/pull/67152
- https://github.com/tgstation/tgstation/pull/67157
- https://github.com/tgstation/tgstation/pull/67220
- https://github.com/tgstation/tgstation/pull/67310
- https://github.com/tgstation/tgstation/pull/67693
- https://github.com/tgstation/tgstation/pull/67781
- https://github.com/tgstation/tgstation/pull/67828
- https://github.com/tgstation/tgstation/pull/68913
- https://github.com/tgstation/tgstation/pull/69194
- https://github.com/tgstation/tgstation/pull/69618
- https://github.com/tgstation/tgstation/pull/70107
- https://github.com/tgstation/tgstation/pull/70286
- https://github.com/tgstation/tgstation/pull/70303
- https://github.com/tgstation/tgstation/pull/71761
- https://github.com/tgstation/tgstation/pull/72258
- https://github.com/tgstation/tgstation/pull/72341
- https://github.com/tgstation/tgstation/pull/72902
- https://github.com/tgstation/tgstation/pull/73683
- https://github.com/tgstation/tgstation/pull/73706
- https://github.com/tgstation/tgstation/pull/76117
- https://github.com/tgstation/tgstation/pull/76673
- https://github.com/tgstation/tgstation/pull/76962
- https://github.com/tgstation/tgstation/pull/77022
- https://github.com/tgstation/tgstation/pull/77212
- https://github.com/tgstation/tgstation/pull/77241
- https://github.com/tgstation/tgstation/pull/77423
- https://github.com/tgstation/tgstation/pull/78734
- https://github.com/tgstation/tgstation/pull/82011
- https://github.com/tgstation/tgstation/pull/82905
- https://github.com/tgstation/tgstation/pull/83425
- https://github.com/tgstation/tgstation/pull/84112
- https://github.com/tgstation/tgstation/pull/84678
- https://github.com/tgstation/tgstation/pull/84806
- https://github.com/tgstation/tgstation/pull/84927
- https://github.com/tgstation/tgstation/pull/86546
- https://github.com/tgstation/tgstation/pull/86825
- https://github.com/tgstation/tgstation/pull/87452
- https://github.com/tgstation/tgstation/pull/87482
- https://github.com/tgstation/tgstation/pull/87649
</details>

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

This only took 20+ hours and a year & a half-worth of pre-reqs.

If you want to see my best assessment of all the work that went into this, uh here you go.

<details>

- https://viewer.diagrams.net/?tags=%7B%7D&lightbox=1&highlight=0000ff&layers=1&nav=1&title=MODSUITS-CHART.drawio#Uhttps%3A%2F%2Fdrive.google.com%2Fuc%3Fid%3D1BNkltT0xfF1m9ZoYlbBgDSVb3TNtfVzn%26export%3Ddownload

![image](https://github.com/user-attachments/assets/4571e5c2-4630-47f4-b29c-a9e701463023)

</details>


I wanted to introduce modsuits here, while also testing keeping hardsuits around for their straight armor upgrade against modsuits. 

At roundstart, it is intended that hardsuits outclass other station-side modsuits.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

Suit Up

https://github.com/user-attachments/assets/18a2e21f-4729-4981-8089-b6a78f8e6fce

Suit Release

https://github.com/user-attachments/assets/b9d801c2-4630-4fb1-b13c-5c7de920a868

Suit up & Module menu demonstration

https://github.com/user-attachments/assets/1d43e0e6-814e-423f-8e44-1a922da8bd01


During equip, the segmented pieces cleanly equip themselves over clothing. Shoes, gloves, suit and hat do not need to be removed.

https://github.com/user-attachments/assets/7e3467d6-68fb-41da-9908-5440a226eeac

Kinesis 


https://github.com/user-attachments/assets/c57c19c1-dd5b-470f-a56b-22c98f7b2921




</details>

## Changelog
:cl: rkz, Fikou, MrMelbert, MTandi, Zonespace27, HatterHat, san7890, dawsonkeyes, mc-oofert, tralezab, SmArtKar, coiax, Ryll/Shaps, Jacquerel, necromanceranne, Xander3395, grungussuss, GPeckman, Holoo-1, OrionTheFox, Momo8289, Wallemations, 
add: modsuits
tweak: all ERT have been converted to use modsuits
tweak: all antagonists that had hardsuits (Nukies, Spacewizard, Uplink Traitors) have been converted to use modsuits
add: adds a loader modsuit for cargo to use. It is not spaceproof and has very little module support, but is good for hauling crates
code: adds unit test to confirm all outfits apply correctly
del: deletes antagonist, ERT, and head of staff hardsuits
refactor: refactored wizard casting to check a clothing flag, instead of a var
tweak: captain's backpack now fits on both MODsuits & suit storage slots
fix: fixes spellcards targeting. Revamps lockon component.
add: adds infiltrator kit, a modsuit that hides your species mutantorgans like wings and antennae, with superior armor, normal runspeed and silent footsteps, in exchange for no spaceproofing.
refactor: refactors ninja gear to all use modsuit modules 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Tsar-Salat committed:
prox

Tsar-Salat committed:
Merge branch 'proximitymonitorrefactor' into modsuits-take5

Tsar-Salat committed:
Time peeps

Tsar-Salat committed:
clown paint

Tsar-Salat committed:
modular cores

Tsar-Salat committed:
wound to bleed

Tsar-Salat committed:
515 and fix tests

Tsar-Salat committed:
kinesis rework

Tsar-Salat committed:
fix issues with nodrop

Tsar-Salat committed:
paper

Tsar-Salat committed:
loader

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/64668

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/64674

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
mining

Tsar-Salat committed:
Merge branch 'modsuits-take5' of https://github.com/Tsar-Salat/BeeStation-Salatland into modsuits-take5

Tsar-Salat committed:
linters

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/64709

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/64857

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/64934

Tsar-Salat committed:
515

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/65000

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/65193

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/65416

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/65559

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/65915

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/65445

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/65733

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/66087

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/65753

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/61138 https://github.com/tgstation/tgstation/pull/62907

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/63755

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/66170

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/66190

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/66196

Tsar-Salat committed:
https://github.com/tgstation/tgstation/pull/66208

Tsar-Salat committed:
techweb

Tsar-Salat committed:
prettier

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge branch 'master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
compiler errors

Tsar-Salat committed:
fix conflicts

Tsar-Salat committed:
declarations

Tsar-Salat committed:
defines

Tsar-Salat committed:
.dme

Tsar-Salat committed:
more .dme

Tsar-Salat committed:
fix unittest

Tsar-Salat committed:
tick unit test

Tsar-Salat committed:
try this

Tsar-Salat committed:
try an early return instead

Tsar-Salat committed:
run list length

Tsar-Salat committed:
oops

Tsar-Salat committed:
deeper we go

Tsar-Salat committed:
this is valid, supposedly

Tsar-Salat committed:
losing my mind

Tsar-Salat committed:
need to check something

Tsar-Salat committed:
syndie box

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
name

Tsar-Salat committed:
congruence

Tsar-Salat committed:
try this

Tsar-Salat committed:
ugh

Tsar-Salat committed:
shielded fix

Tsar-Salat committed:
genericize modsuit parts

https://github.com/tgstation/tgstation/pull/66520

Tsar-Salat committed:
overslot fixes

https://github.com/tgstation/tgstation/pull/66574

Tsar-Salat committed:
any slot we want

https://github.com/tgstation/tgstation/pull/66667

Tsar-Salat committed:
module update

https://github.com/tgstation/tgstation/pull/67042

Tsar-Salat committed:
kinesis buff

https://github.com/tgstation/tgstation/pull/67157

Tsar-Salat committed:
out manager support

https://github.com/tgstation/tgstation/pull/67310

Tsar-Salat committed:
ore bag support

https://github.com/tgstation/tgstation/pull/67781

Tsar-Salat committed:
9/02/2022 modsuit fixes

https://github.com/tgstation/tgstation/pull/69618

Tsar-Salat committed:
suit storage

https://github.com/tgstation/tgstation/pull/70107

Tsar-Salat committed:
module pinning

Tsar-Salat committed:
modsuit AI's count for objectives

https://github.com/tgstation/tgstation/pull/72902

Tsar-Salat committed:
medical modsuit update

https://github.com/tgstation/tgstation/pull/69194

Tsar-Salat committed:
status readout

https://github.com/tgstation/tgstation/pull/76117

Tsar-Salat committed:
new modsuit ui

Tsar-Salat committed:
fix on/off

https://github.com/tgstation/tgstation/pull/72341

Tsar-Salat committed:
modsuit adapter shell

https://github.com/tgstation/tgstation/pull/70286

Tsar-Salat committed:
WIP on modsuits-take5

Tsar-Salat committed:
unhardcode modsuit parts

https://github.com/tgstation/tgstation/pull/82905,

Tsar-Salat committed:
casting

Tsar-Salat commented on code/__DEFINES/is_helpers.dm:
```diff
@@ -77,6 +77,8 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
 
 #define ishuman(A) (istype(A, /mob/living/carbon/human))
 
+#define ishumantesting(A) (istype(A, /mob/living/carbon/human/consistent) || istype(A, /mob/living/carbon/human/dummy))
```
Specifically when we need to run something in a testing enviroment that wouldnt work in a real environment 
(the clockie gear shocks nonclockies for instance when worn)

Tsar-Salat commented on code/datums/action.dm:
```diff
@@ -148,6 +151,27 @@
 	UpdateButtonIcon()
 
+//Adds our action button to the screen of another player
+/datum/action/proc/Share(mob/freeloader)
```
Needed for AI's to control modsuits.

Tsar-Salat committed:
circuit issue

Tsar-Salat committed:
nuclear option

Tsar-Salat committed:
limbless rejoice

https://github.com/tgstation/tgstation/pull/86825 https://github.com/tgstation/tgstation/pull/73538

Tsar-Salat committed:
fix overlay

Tsar-Salat committed:
GRRRR

Tsar-Salat committed:
fix up segmentation issues

https://github.com/tgstation/tgstation/pull/87482

Tsar-Salat committed:
pinner?

Tsar-Salat committed:
processing fix

Tsar-Salat committed:
Revert "pinner?"

This reverts commit ea95eac19e741f5cef01b6645a804fab585d2132.

Tsar-Salat committed:
fix bad del

Tsar-Salat committed:
mapedits

Tsar-Salat committed:
oops

Tsar-Salat committed:
rad fix

Tsar-Salat committed:
fix status_traits runtime

Tsar-Salat committed:
add lights

Tsar-Salat committed:
engineering tweaks

Tsar-Salat committed:
fix kinesis

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
merge skew oopsie

Tsar-Salat committed:
looping sound, map, and outfit

Tsar-Salat committed:
oops

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
trait refs

https://github.com/tgstation/tgstation/pull/87726

Tsar-Salat committed:
Revert "trait refs"

This reverts commit a43239cc40184d3002329d8386ce06d4bc1d293d.

Tsar-Salat commented on code/modules/mod/modules/modules_antag.dm:
```diff
@@ -0,0 +1,440 @@
+	for(var/armor_type in removed_armor)
+		removed_armor[armor_type] = -removed_armor[armor_type]
+	for(var/obj/item/part as anything in mod.get_parts(all = TRUE))
+		part.armor = part.armor.modifyRating(arglist(removed_armor))
```
> hi if you have any questions i can answer them here

@Fikou haha, might have to take you up on it.

Did you ever have an issue with module deactivation on equip like this? 

Specifically I seem to be running into this armorlist proc issue with the armor booster. (same one killing my workflow) We are still on the list-based armor datums rather than TG with subtyped armor.

![image](https://github.com/user-attachments/assets/d4881e6d-1f25-4ba6-9ff6-f68ece733e8b)

It's the same code as the `on_activation()`, its just copied to a new list, set to its inverse, then converted from a list to params by the `arglist()`.

Atleast, thats my understanding of it, I'm still learning list manipulation. Do you have any thoughts?

Tsar-Salat replied:
Sorry for bothering you Fikou, I figured it out. I don't know how you got it to work in the original Modsuits, I just had to do some extra list magic to get it to work.

Thank you for your time regardless

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
le fix

Tsar-Salat committed:
kar fix

https://github.com/tgstation/tgstation/pull/87649

Tsar-Salat committed:
ethereal heart business

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
kinetic acc

Tsar-Salat committed:
oops

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

PowerfulBacon commented on code/__byond_version_compat.dm:
```diff
@@ -28,7 +28,7 @@
 // That being said, if you really are ready, you can give YES_I_WANT_515 to TGS.
 #if !defined(YES_I_WANT_515) && DM_VERSION >= 515
-#error We do not yet completely support BYOND 515.
+#warn We do not yet completely support BYOND 515.
```
This needs to be reverted

Tsar-Salat replied:
This is when this error broke StrongDMM compilation

975bc70

PowerfulBacon commented on code/game/objects/items/storage/belt.dm:
```diff
@@ -738,7 +755,7 @@
 	new /obj/item/holosign_creator/janibarrier(src)
 	new /obj/item/melee/flyswatter(src)
-	new /obj/item/reagent_containers/cup/bucket(src)
+	new /obj/item/melee/baton/loaded(src)
```
Is this intentional?

Tsar-Salat replied:
Yes, the belt name is a misnomer, its only used by janitor ERT.

Due to Ruko's storage rebalancing, I've had to move a lot of items around so they actually fit when the outfit is equipped. In this case, batons are a large item that didnt fit anywhere else on the janitor ert, so its now in the belt.

Ive changed the name of the belt subtype so its easily known its for ERT. 56c5439

PowerfulBacon commented on code/modules/mob/living/silicon/robot/robot_modules.dm:
```diff
@@ -722,7 +720,7 @@
 	moduleselect_icon = "malf"
 	can_be_pushed = FALSE
-	magpulsing = TRUE
+	//module_traits = list(TRAIT_PUSHIMMUNE, TRAIT_NEGATES_GRAVITY)
```
Do we want this commented out?

Tsar-Salat replied:
We did not support per-module traits on borgs yet, which is now fixed.

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
cooldown removal

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
conversion & adds rad protection to engineering modsuits

Tsar-Salat committed:
hmm, thats supposed to be a number...

Tsar-Salat committed:
No wonder the modules didnt fucking work...

Tsar-Salat committed:
update for gravity pr

Tsar-Salat committed:
trying to fix visors...

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
fix ert name

Tsar-Salat committed:
add support for module traits

https://github.com/tgstation/tgstation/pull/55769

Tsar-Salat committed:
revert warn

Tsar-Salat committed:
fix chameleon module

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Dejaku51 commented on code/__DEFINES/access.dm:
```diff
@@ -115,3 +115,12 @@
+/// Used in ID card access adding procs. Will forcefully add all accesses.
+#define FORCE_ADD_ALL 2
+/// Used in ID card access adding procs. Will stack trace on fail.
+#define ERROR_ON_FAIL 3
```
nothing except ERROR_ON_FAIL is used?

Tsar-Salat replied:
3a8fa55

Dejaku51 commented on code/game/objects/items/cards_ids.dm:
```diff
@@ -206,6 +206,9 @@
 	add_fingerprint(user)
 
+/obj/item/card/id/proc/set_access(list/new_access_list, mode = ERROR_ON_FAIL)
+	return
```
theres isn't any override to this so set_access always just return

Tsar-Salat replied:
This is me trying to bypass TG's ID system

Tsar-Salat replied:
3a8fa55

Dejaku51 commented on code/__DEFINES/cargo.dm:
```diff
@@ -46,6 +46,8 @@
 
 #define SUPPLYPOD_X_OFFSET -16
 
+#define CARGO_CRATE_VALUE 200
```
cargo crates are worth 500 as seen in code/modules/cargo/exports/large_objects.dm. also if you're gonna add this define, crate export value should be set to the define

Tsar-Salat replied:
3a8fa55

Dejaku51 commented on code/game/atoms_movable.dm:
```diff
@@ -253,16 +253,18 @@
+	var/mob/living/old_pulling = pulling
+	pulling = null
+	SEND_SIGNAL(old_pulling, COMSIG_ATOM_NO_LONGER_PULLED, src)
+	SEND_SIGNAL(src, COMSIG_ATOM_NO_LONGER_PULLING, old_pulling)
```
signal gets sent but nothing registers it

Tsar-Salat replied:
3a8fa55

Dejaku51 commented on code/__DEFINES/dcs/signals/signals_datum/signals_datum.dm:
```diff
@@ -65,6 +65,11 @@
+#define COMSIG_DISPENSERBOT_ADD_ITEM "moneybot_add_item"
+
+/// Called in /obj/structure/dispenserbot/remove_item(). (obj/item/to_remove)
+#define COMSIG_DISPENSERBOT_REMOVE_ITEM "moneybot_remove_item"
```
both defines unused

Tsar-Salat replied:
3a8fa55

Dejaku51 commented on code/__DEFINES/dcs/signals/signals_lighting.dm:
```diff
@@ -38,3 +38,6 @@
 
+/// from base of atom/movable/Process_Spacemove(): (movement_dir)
+#define COMSIG_MOVABLE_SPACEMOVE "spacemove"
+	#define COMSIG_MOVABLE_STOP_SPACEMOVE (1<<0)
```
is this the right file for that?

Tsar-Salat replied:
3a8fa55

Dejaku51 commented on code/__DEFINES/dcs/signals/signals_mob/signals_silicon.dm:
```diff
@@ -7,3 +7,7 @@
 #define COMSIG_MMI_SET_BRAINMOB "mmi_set_brainmob"
+
+/// from base of /obj/item/slimepotion/speed/afterattack(): (obj/target, /obj/src, mob/user)
+#define COMSIG_SPEED_POTION_APPLIED "speed_potion"
```
also probably bad file for this

Tsar-Salat replied:
3a8fa55

Dejaku51 commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,315 @@
+			"<span class='notice'>[part] deploy[part.p_s()] with a mechanical hiss.</span>",
+			"<span class='hear'>You hear a mechanical hiss.</span>")
+		playsound(src, 'sound/mecha/mechmove03.ogg', 25, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+		SEND_SIGNAL(src, COMSIG_MOD_PART_DEPLOYED, user, part)
```
unused signal

Dejaku51 commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,315 @@
+		if(!QDELING(wearer) && !wearer.equip_to_slot_if_possible(overslot, overslot.slot_flags, qdel_on_fail = FALSE, disable_warning = TRUE))
+			wearer.dropItemToGround(overslot, force = TRUE, silent = TRUE)
+	wearer.update_clothing(slot_flags)
+	SEND_SIGNAL(src, COMSIG_MOD_PART_RETRACTED, user, part)
```
unused signal

Dejaku51 commented on code/modules/mod/mod_actions.dm:
```diff
@@ -0,0 +1,196 @@
+	desc = "Quickly activate [linked_module]."
+	icon_icon = linked_module.icon
+	button_icon_state = linked_module.icon_state
+	RegisterSignal(linked_module, COMSIG_MODULE_ACTIVATED, PROC_REF(on_module_activate))
```
unused signal (nothing sends it)

Tsar-Salat replied:
f7d618e

Dejaku51 commented on code/modules/mod/mod_actions.dm:
```diff
@@ -0,0 +1,196 @@
+	button_icon_state = linked_module.icon_state
+	RegisterSignal(linked_module, COMSIG_MODULE_ACTIVATED, PROC_REF(on_module_activate))
+	RegisterSignal(linked_module, COMSIG_MODULE_DEACTIVATED, PROC_REF(on_module_deactivate))
+	RegisterSignal(linked_module, COMSIG_MODULE_USED, PROC_REF(on_module_use))
```
unregister those signals on destroy

Tsar-Salat replied:
f7d618e

Dejaku51 reviewed (changes requested):
thats all for now, will continue another day

Dejaku51 commented on code/datums/ai/objects/mod.dm:
```diff
@@ -0,0 +1,48 @@
+/// An AI controller for the MODsuit pathfinder module. It's activated by implant and attaches itself to the user.
+/datum/ai_controller/mod
```
suit fails to go through any doors (if no access, open the door and says connection lost)

Tsar-Salat replied:
Yea I cant really improve this without substantial JPS reworking

Dejaku51 replied:
not happy but its good enough for this PR

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
dejaku fix 1

Tsar-Salat committed:
dejaku fix 2

Tsar-Salat committed:
dejaku fix 3

Tsar-Salat committed:
dejaku fix 4

Tsar-Salat committed:
unused

Dejaku51 commented on code/__DEFINES/dcs/signals/signals_datum/signals_datum.dm:
```diff
@@ -65,7 +65,6 @@
-
 #define COMSIG_GREYSCALE_CONFIG_REFRESHED "greyscale_config_refreshed"
 
 
```
we don't have dispenserbots, remove those signals

Tsar-Salat replied:
what?

Dejaku51 replied:
ffs another one of the git moment showing me code that isn't there anymore

Dejaku51 commented on code/__DEFINES/inventory.dm:
```diff
@@ -171,7 +173,7 @@ GLOBAL_LIST_INIT(security_hardsuit_allowed, typecacheof(list(
 	/obj/item/restraints/handcuffs,
 	/obj/item/tank/internals)))
 
-GLOBAL_LIST_INIT(detective_vest_allowed, typecacheof(list(
```
was that accidental?

Tsar-Salat replied:
is_type_in_list aren't typecaches

Dejaku51 commented on code/__DEFINES/inventory.dm:
```diff
@@ -190,7 +192,7 @@ GLOBAL_LIST_INIT(detective_vest_allowed, typecacheof(list(
 	/obj/item/tank/internals/emergency_oxygen,
 	/obj/item/tank/internals/plasmaman)))
 
-GLOBAL_LIST_INIT(security_vest_allowed, typecacheof(list(
```
was that accidental?

Tsar-Salat replied:
is_type_in_list aren't typecaches

Dejaku51 commented on code/__DEFINES/inventory.dm:
```diff
@@ -206,7 +208,7 @@ GLOBAL_LIST_INIT(security_vest_allowed, typecacheof(list(
 	/obj/item/tank/internals/emergency_oxygen,
 	/obj/item/tank/internals/plasmaman)))
 
-GLOBAL_LIST_INIT(security_wintercoat_allowed, typecacheof(list(
```
was that accidental?

Tsar-Salat replied:
ditto

Dejaku51 commented on code/__DEFINES/is_helpers.dm:
```diff
@@ -286,3 +290,16 @@ GLOBAL_LIST_INIT(book_types, typecacheof(list(
+#define is_scientist_job(job_type) (istype(job_type, /datum/job/scientist))
+#define is_security_officer_job(job_type) (istype(job_type, /datum/job/security_officer))
+#define is_research_director_job(job_type) (istype(job_type, /datum/job/research_director))
+#define is_unassigned_job(job_type) (istype(job_type, /datum/job/unassigned))
```
those areen't used anywhere

Tsar-Salat replied:
the clown one is used, but it would be rather bad of me to make exactly one istype define without the others

Dejaku51 replied:
sure

Dejaku51 commented on code/__DEFINES/mobs.dm:
```diff
@@ -396,6 +396,12 @@ GLOBAL_LIST_INIT(available_random_trauma_list, list(
+#define FLASH_PROTECTION_SENSITIVE -1
+#define FLASH_PROTECTION_NONE 0
+#define FLASH_PROTECTION_FLASH 1
+#define FLASH_PROTECTION_WELDER 2
```
if you're adding those, should probably make everything use it (simple find & replace should do itt)

Tsar-Salat replied:
f4368d2

Dejaku51 commented on code/__DEFINES/traits/declarations.dm:
```diff
@@ -72,8 +72,10 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
+#define TRAIT_NODISMEMBER "dismember_immunity"
+#define TRAIT_NOFIRE "nonflammable"
+/// Prevents plasmamen from self-igniting
+#define TRAIT_NOSELFIGNITION "no_selfignition"
```
you removed the space from TRAIT_NODISMEMBER and TRAIT_NOFIRE. also TRAIT_NOSELFIGNITION isn't checked anywhere

Tsar-Salat replied:
f4368d2

Dejaku51 commented on code/__DEFINES/traits/declarations.dm:
```diff
@@ -86,14 +88,20 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
+#define TRAIT_FASTMED "fast_med_use"
 #define TRAIT_NOBREATH			"no_breath"
+/// Like antimagic, but doesn't block the user from casting
+#define TRAIT_ANTIMAGIC_NO_SELFBLOCK "anti_magic_no_selfblock"
```
also isn't checked anywhere

Tsar-Salat replied:
f727e12

Dejaku51 commented on code/__DEFINES/traits/declarations.dm:
```diff
@@ -86,14 +88,20 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 /// Stops all slipping and sliding from ocurring
 #define TRAIT_NOSLIPALL "noslip_all"
+/// Prevents some severe head injuries being sustained from heavy collisions or blunt force injuries.
+#define TRAIT_HEAD_INJURY_BLOCKED "head_injury_blocked"
```
also isn't checked anywhere. not even sure if its for our med

Tsar-Salat replied:
from deleted security module, will remove

Tsar-Salat replied:
f4368d2

Dejaku51 commented on code/__DEFINES/traits/declarations.dm:
```diff
@@ -116,9 +124,19 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
+/// Gives us turf, mob and object vision through walls
+#define TRAIT_XRAY_VISION "xray_vision"
+/// Gives us mob vision through walls and slight night vision
+#define TRAIT_THERMAL_VISION "thermal_vision"
```
tabs were removed

Tsar-Salat replied:
There is no requirement to use tabs.

![image](https://github.com/user-attachments/assets/dc17fa72-b604-40c5-97bf-bfdb58a4c762)

I wrote these defines and they are the cleanest in the whole file

Dejaku51 replied:
the problem is that they're different to the rest of the file

![image](https://github.com/user-attachments/assets/e6091acb-7e2b-4733-ba18-e10caae07cbf)

Dejaku51 commented on code/__DEFINES/traits/declarations.dm:
```diff
@@ -116,9 +124,19 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
+/// Gives us Night vision
+#define TRAIT_TRUE_NIGHT_VISION "true_night_vision"
+/// Lets us scan reagents
+#define TRAIT_REAGENT_SCANNER "reagent_scanner"
```
should probably implement it on normal mesons etc

Tsar-Salat replied:
Mesons already use TRAIT_MESON_VISION

Dejaku51 replied:
nope

![image](https://github.com/user-attachments/assets/d330ac6f-6447-4851-9cf6-9251904b325f)

Tsar-Salat replied:
🥹

Tsar-Salat replied:
86401a2

Dejaku51 commented on code/__DEFINES/traits/declarations.dm:
```diff
@@ -131,19 +149,24 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
+#define TRAIT_QUICK_CARRY "quick-carry"
+/// We place people into a fireman carry especially quickly compared to quick_carry
+#define TRAIT_QUICKER_CARRY "quicker-carry"
+#define TRAIT_QUICK_BUILD "quick-build"
```
also bad tabs

Tsar-Salat replied:
what does this mean???

Dejaku51 commented on code/__DEFINES/traits/declarations.dm:
```diff
@@ -131,19 +149,24 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_DIAGNOSTIC_HUD "diag_hud"
-#define TRAIT_MEDIBOTCOMINGTHROUGH "medbot" //Is a medbot healing you
+/// Is a medbot healing you
+#define TRAIT_MEDIBOTCOMINGTHROUGH "medbot"
```
also bad tabs

Tsar-Salat replied:
what? There is no tabs here

Dejaku51 commented on code/_globalvars/bitfields.dm:
```diff
@@ -206,16 +206,16 @@ DEFINE_BITFIELD(clothing_flags, list(
 	"SNUG_FIT" = SNUG_FIT,
 	"EFFECT_HAT" = EFFECT_HAT,
-	"SCAN_REAGENTS" = SCAN_REAGENTS,
-	"SCAN_BOOZEPOWER" = SCAN_BOOZEPOWER,
```
why were tthose removed? they still exist

Tsar-Salat replied:
forgor to delete

Tsar-Salat replied:
f4368d2

Dejaku51 commented on code/datums/diseases/advance/symptoms/heal.dm:
```diff
@@ -87,7 +87,7 @@
 	suffixes = list(" Coma")
 	var/deathgasp = FALSE
-	var/stabilize = FALSE
+	var/stabilize = TRUE
```
why the change?

Tsar-Salat replied:
idk

Tsar-Salat replied:
f4368d2

Dejaku51 commented on code/datums/looping_sounds/machinery_sounds.dm:
```diff
@@ -90,6 +90,19 @@
+/datum/looping_sound/gravgen/kinesis
+	volume = 20
+	falloff_distance = 2
+	falloff_exponent = 5
```
should probably add a use to /datum/looping_sound/gravgen if its being added

Tsar-Salat replied:
I tried it, it breaks and I don't want to bloat this PR any more.

Dejaku51 commented on code/datums/components/shielded.dm:
```diff
@@ -239,15 +257,12 @@
-		return
 
 	max_integrity += recharge_rune.added_shield
 	adjust_charge(recharge_rune.added_shield)
```
could probably make it so recharge rune is a component/element to avoid adding another variable thats barely used

Tsar-Salat replied:
recharge ruin can be anything. It just has to have added_shield var saying how much its adding, and match the correct recharge_path.

Modularizing this any more doesn't really make sense.

Dejaku51 commented on code/game/objects/items/tanks/watertank.dm:
```diff
@@ -96,6 +97,7 @@
 
 	if(QDELETED(noz))
 		noz = make_noz()
+		RegisterSignal(noz, COMSIG_MOVABLE_MOVED, PROC_REF(noz_move))
```
maybe instead, register signal in make_noz()

Tsar-Salat replied:
the parent and the subtype both use the signal, so this file would still have two RegisterSignal(MOVABLE_MOVED)

not much point

Dejaku51 commented on code/game/objects/items/tanks/watertank.dm:
```diff
@@ -483,14 +483,8 @@
-	..()
-
 /obj/item/extinguisher/mini/nozzle/attack_self(mob/user)
+	//var/uses_pack = istype(tank, /obj/item/watertank/atmos)
```
why

Tsar-Salat replied:
b7d0842

I left it commented for some reason that was important, but I forget what it was lol

Dejaku51 commented on code/game/objects/structures/crates_lockers/closets/bodybag.dm:
```diff
@@ -60,7 +60,7 @@
 		mouse_drag_pointer = MOUSE_INACTIVE_POINTER
 
-/obj/structure/closet/body_bag/close()
+/obj/structure/closet/body_bag/close(mob/living/user)
```
unused argument

Tsar-Salat replied:
is there an issue with passing unused arguments if the parent uses it?

I guess I could refrain from it, but theres certainly enough procs that just pass the same args as their parents without using them.

Dejaku51 replied:
eh not really an issue

Dejaku51 commented on code/modules/awaymissions/corpse.dm:
```diff
@@ -366,8 +366,8 @@ CREATION_TEST_IGNORE_SELF(/obj/effect/mob_spawn)
-/obj/effect/mob_spawn/human/miner/rig
-	outfit = /datum/outfit/job/miner/equipped/hardsuit
+/obj/effect/mob_spawn/human/miner/mod
+	outfit = /datum/outfit/job/engineer/mod
```
miner but engineer outfit?

Tsar-Salat replied:
f4368d2

Dejaku51 commented on code/modules/client/client_colour.dm:
```diff
@@ -206,6 +206,9 @@
 	addtimer(CALLBACK(src, PROC_REF(update_colour), list(1,0,0,0.8,0.2,0, 0.8,0,0.2,0.1,0,0), 10, SINE_EASING|EASE_OUT), 1)
 
+/datum/client_colour/rave
+	priority = PRIORITY_LOW
```
will that do anything without other variables?

Tsar-Salat replied:
client colours do not have to have their properties defined. They can be modified by procs.

This subtype is used in the rave, but the only important thing is setting its priority low.

Dejaku51 replied:
oke

Dejaku51 commented on code/modules/clothing/outfits/standard.dm:
```diff
@@ -33,6 +34,39 @@
+	W.registered_name = H.real_name
+	W.update_label()
+	W.update_icon()
+*/
```
that code will probably stay commented forever so it should be removed entirely

Tsar-Salat replied:
f4368d2

Dejaku51 commented on code/modules/clothing/spacesuits/_spacesuits.dm:
```diff
@@ -48,6 +48,7 @@
 	allowed = list(
 		/obj/item/flashlight,
 		/obj/item/tank/internals,
+		/obj/item/tank/jetpack/oxygen/captain,
```
captains jetpack???

Tsar-Salat replied:
Yes. With Modsuits it's a useless because you can't put it on your back or in the hardsuit. (Since we are removing head hardsuits)

Putting it in the suit storage slot makes what's basically a relic more usable.

Dejaku51 replied:
disregard

Dejaku51 reviewed (changes requested):
did some more

Tsar-Salat committed:
requested changes

Tsar-Salat committed:
bad idea

Tsar-Salat committed:
forgot about you

Tsar-Salat committed:
unused

Tsar-Salat committed:
Merge branch 'master' into modsuits-take5

Dejaku51 commented on code/modules/wiremod/shell/module.dm:
```diff
@@ -0,0 +1,246 @@
+	display_name = "MOD action"
+	desc = "Represents an action the user can take when wearing the MODsuit."
+	required_shells = list(/obj/item/mod/module/circuit)
+*/
```
why is all of this commented out?

Tsar-Salat replied:
This is desirable that I want to include it, but our circuit code is incompatible with it.

The incompatibilities are minor, but I do not know enough about circuits to make it workable. Ive spent atleast 3 hours on a separate branch with no progress.

Dejaku51 commented on code/modules/wiremod/shell/brain_computer_interface.dm:
```diff
@@ -12,9 +12,12 @@
 	. = ..()
 
+	var/obj/item/integrated_circuit/circuit = new(src)
+	circuit.add_component(new /obj/item/circuit_component/equipment_action/bci(null, "One"))
```
why do you make them start with a circuit? it makes them unable to have cell inserted

Dejaku51 replied:
also why does it start with bci action preequipped

Tsar-Salat replied:
The actions are now part of the circuit, this way you can remove the actions if you dont want them. (Not unremovable)

Dejaku51 replied:
they were like that before, we changed that specifically so players can just just ones they need

Dejaku51 replied:
with your current thing, you 
- print BCI shell
- assemble BCI shell
- take circuit out of BCI shell
- put battery in BCI shell
- put circuit back in BCI shell


also its snowflakey to make BCIi shell the only one that starts with circuit

Tsar-Salat replied:
37a0a42

Dejaku51 commented on code/modules/wiremod/components/abstract/equipment_action.dm:
```diff
@@ -0,0 +1,65 @@
+		"Shield",
+		"Skull",
+		"Sleep",
+		"Wireless",
```
most of these don't exist in icons\hud\actions\actions_items.dmi

Tsar-Salat replied:
89330d5523976cd83abe3dc7becf7e8517bdd2e8

Dejaku51 commented on code/modules/surgery/organs/augments_chest.dm:
```diff
@@ -175,6 +175,12 @@
+	SIGNAL_HANDLER
+
+	if(on && movement_dir)
+		return COMSIG_MOVABLE_STOP_SPACEMOVE
```
signal isn't registered anywhere

Tsar-Salat replied:
it doesnt need to be? It's part of Process_Spacemove()

Dejaku51 replied:
why is it a signal handler then?

Tsar-Salat replied:
every proc/move_react() has a SIGNAL_HANDLER

Tsar-Salat replied:
actually I think I just forgot

Tsar-Salat replied:
05642c1

Dejaku51 commented on code/modules/mod/modules/modules_maint.dm:
```diff
@@ -0,0 +1,334 @@
+/obj/item/mod/module/atrocinator/on_deactivation(display_message = TRUE, deleting = FALSE)
+	if(!deleting)
+		playsound(src, 'sound/effects/curseattack.ogg', 50)
+	qdel(mod.wearer.RemoveElement(/datum/element/forced_gravity, NEGATIVE_GRAVITY))
```
that looks cursed, does RemoveElement even return anything for it to qdel?

Tsar-Salat replied:
I guess not

Tsar-Salat replied:
89330d5523976cd83abe3dc7becf7e8517bdd2e8

Dejaku51 commented on code/modules/mod/modules/modules_maint.dm:
```diff
@@ -0,0 +1,334 @@
+	investigate_log("has flown off into space due to the [src].", INVESTIGATE_DEATHS)
+	mod.wearer.Stun(FLY_TIME, ignore_canstun = TRUE)
+	animate(mod.wearer, FLY_TIME, pixel_y = 256, alpha = 0)
+	QDEL_IN(mod.wearer, FLY_TIME)
```
fun

Tsar-Salat replied:
Whats fun on station isnt good for lavaland

Dejaku51 reviewed (changes requested):
that should be all

Tsar-Salat committed:
address reviews

Tsar-Salat committed:
For you, Meson! Not for me!

Tsar-Salat committed:
antimagic check

Tsar-Salat committed:
remove from atmosians (explo, mining, and atmos are now the unmodded)

Tsar-Salat committed:
spacemove

Tsar-Salat committed:
revert bci

Tsar-Salat committed:
hnnnnng

Tsar-Salat committed:
fix mesons

Tsar-Salat committed:
examinate

Tsar-Salat committed:
.

Tsar-Salat committed:
pathfinder and individual suit fixes

https://github.com/tgstation/tgstation/pull/84927

Tsar-Salat committed:
autoupdate

Tsar-Salat committed:
balance and add loaders/medical modsuits to roundstart station

Tsar-Salat committed:
forcestanding

Tsar-Salat committed:
minor untangle

Tsar-Salat committed:
lower slowdown

Tsar-Salat committed:
adds pathfinder plus, the suit that loves you

Tsar-Salat committed:
domineering AIs, better be diplomatic.

Tsar-Salat committed:
lower highest tiers by 1

Tsar-Salat committed:
dont lock it if its still open

Tsar-Salat committed:
oops

Tsar-Salat committed:
supply

Tsar-Salat committed:
maybe?

Tsar-Salat committed:
WIP on hotkeyshit

Tsar-Salat committed:
.

Tsar-Salat committed:
fix drop key

Tsar-Salat committed:
actions

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
try this

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
.

Tsar-Salat committed:
span conversions

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

Tsar-Salat committed:
unlock springlock functionality

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into modsuits-take5

# Conflicts:
#	_maps/map_files/generic/CentCom.dmm
#	code/__DEFINES/actions.dm
#	code/__DEFINES/dcs/signals/signals_datum/signals_datum.dm
#	code/__DEFINES/dcs/signals/signals_mob/signals_living.dm
#	code/__DEFINES/dcs/signals/signals_mob/signals_mob.dm
#	code/__DEFINES/obj_flags.dm
#	code/__DEFINES/traits/sources.dm
#	code/__HELPERS/roundend.dm
#	code/_globalvars/bitfields.dm
#	code/_globalvars/traits/_traits.dm
#	code/_onclick/hud/action_button.dm
#	code/controllers/subsystem/traumas.dm
#	code/controllers/subsystem/vote.dm
#	code/datums/action.dm
#	code/datums/components/manual_blinking.dm
#	code/datums/components/manual_breathing.dm
#	code/datums/elements/cleaning.dm
#	code/datums/martial/krav_maga.dm
#	code/datums/martial/wrestling.dm
#	code/datums/view.dm
#	code/game/machinery/porta_turret/portable_turret.dm
#	code/game/objects/items/chrono_eraser.dm
#	code/game/objects/items/devices/multitool.dm
#	code/game/objects/items/tanks/watertank.dm
#	code/game/objects/structures/popout_cake.dm
#	code/modules/antagonists/changeling/changeling_power.dm
#	code/modules/antagonists/changeling/powers/augmented_eyesight.dm
#	code/modules/antagonists/changeling/powers/tiny_prick.dm
#	code/modules/antagonists/traitor/equipment/Malf_Modules.dm
#	code/modules/antagonists/wizard/equipment/spellbook.dm
#	code/modules/clothing/chameleon.dm
#	code/modules/clothing/gloves/miscellaneous.dm
#	code/modules/clothing/spacesuits/chronosuit.dm
#	code/modules/clothing/spacesuits/hardsuit.dm
#	code/modules/clothing/suits/wiz_robe.dm
#	code/modules/detectivework/scanner.dm
#	code/modules/mining/lavaland/necropolis_chests.dm
#	code/modules/mob/living/carbon/human/inventory.dm
#	code/modules/mob/living/carbon/human/species_types/diona.dm
#	code/modules/mob/living/carbon/human/species_types/dullahan.dm
#	code/modules/mob/living/carbon/human/species_types/plasmamen.dm
#	code/modules/mob/living/carbon/human/species_types/psyphoza.dm
#	code/modules/mob/living/carbon/human/species_types/pumpkin_man.dm
#	code/modules/mob/living/carbon/human/species_types/vampire.dm
#	code/modules/mob/living/living.dm
#	code/modules/mob/living/silicon/ai/ai.dm
#	code/modules/mob/living/silicon/pai/pai.dm
#	code/modules/mob/living/silicon/robot/robot.dm
#	code/modules/mob/living/simple_animal/hostile/bosses/boss.dm
#	code/modules/mob/living/simple_animal/hostile/bosses/paperwizard.dm
#	code/modules/mob/living/simple_animal/hostile/giant_spider.dm
#	code/modules/mob/living/simple_animal/hostile/goose.dm
#	code/modules/mob/living/simple_animal/hostile/mecha_pilot.dm
#	code/modules/mob/living/simple_animal/hostile/retaliate/nymph.dm
#	code/modules/mob/living/simple_animal/hostile/space_dragon.dm
#	code/modules/mob/mob.dm
#	code/modules/projectiles/gun.dm
#	code/modules/research/xenobiology/crossbreeding/_clothing.dm
#	code/modules/ruins/spaceruin_code/hilbertshotel.dm
#	code/modules/spells/__DEFINES/spell.dm
#	code/modules/surgery/dental_implant.dm
#	code/modules/surgery/organs/augments_internal.dm
#	code/modules/surgery/organs/heart.dm
#	code/modules/surgery/organs/vocal_cords.dm
#	code/modules/surgery/organs/wings.dm
#	code/modules/vehicles/mecha/combat/durand.dm
#	code/modules/vehicles/mecha/mecha_actions.dm
#	code/modules/vehicles/mecha/working/clarke.dm
#	code/modules/vehicles/vehicle_actions.dm
#	code/modules/wiremod/shell/brain_computer_interface.dm
#	tgui/packages/tgui/components/LabeledList.tsx

Tsar-Salat committed:
MOD action button resprite

https://github.com/tgstation/tgstation/pull/70988

Tsar-Salat committed:
Merge branch 'master' into modsuits-take5

Tsar-Salat committed:
PAI update

https://github.com/tgstation/tgstation/pull/77212 https://github.com/tgstation/tgstation/pull/82011

Tsar-Salat committed:
Readd rclick features

Tsar-Salat committed:
fix

Tsar-Salat committed:
fix spellcards and kinesis

https://github.com/tgstation/tgstation/pull/73706

Tsar-Salat committed:
fix messed up rmb interaction

Tsar-Salat committed:
prettier

Tsar-Salat committed:
fix cc

Tsar-Salat committed:
why did I do that

Tsar-Salat committed:
fix huds, fix bad action buttons

https://github.com/tgstation/tgstation/pull/88326 https://github.com/tgstation/tgstation/pull/87726 https://github.com/tgstation/tgstation/pull/88369

Tsar-Salat committed:
re-evaluate emp shields, fix adminsuit theme

Tsar-Salat committed:
armor adjustments, removes erroneous ethereal heart desc

Tsar-Salat committed:
fix anomaly cores

Tsar-Salat committed:
fix modpaint

Tsar-Salat committed:
tech too

Tsar-Salat committed:
Infiltrator. Hides features, not spaceproof, great defense.

https://github.com/tgstation/tgstation/pull/70303 https://github.com/tgstation/tgstation/pull/86546 https://github.com/tgstation/tgstation/pull/68913 https://github.com/tgstation/tgstation/pull/84112

Tsar-Salat committed:
civilian modsuit

https://github.com/tgstation/tgstation/pull/84806

Tsar-Salat committed:
Ninja modsuit (unimplemented)

https://github.com/tgstation/tgstation/pull/67220 https://github.com/tgstation/tgstation/pull/84678 https://github.com/tgstation/tgstation/pull/77423

Tsar-Salat committed:
hat stabilizer

https://github.com/tgstation/tgstation/pull/63795 https://github.com/tgstation/tgstation/pull/76962

Tsar-Salat committed:
reenable pepper shoulders

Tsar-Salat committed:
projectile dampeners & megaphone

Tsar-Salat committed:
active sonar

https://github.com/tgstation/tgstation/pull/67828 https://github.com/tgstation/tgstation/pull/78734

Tsar-Salat committed:
adv emp shield

https://github.com/tgstation/tgstation/pull/67152

Tsar-Salat committed:
lol

Tsar-Salat committed:
clear

Tsar-Salat committed:
gruhhh

Tsar-Salat committed:
fix armor comparisions

Tsar-Salat committed:
fix uplink kit giving restricted modsuit

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

Tsar-Salat committed:
fix timeline

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	code/__DEFINES/dcs/signals/signals_mob/signals_mob.dm

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	beestation.dme
#	code/__DEFINES/dcs/signals/signals_atom/signals_atom.dm
#	code/__DEFINES/dcs/signals/signals_mob/signals_living.dm
#	code/__DEFINES/dcs/signals/signals_obj/signals_item/signals_item.dm
#	code/__DEFINES/obj_flags.dm
#	code/__DEFINES/traits/declarations.dm
#	code/game/objects/items.dm
#	code/modules/cargo/bounties/core.dm
#	code/modules/research/techweb/all_nodes.dm

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	code/datums/components/storage/storage.dm
#	code/game/atoms.dm
#	code/game/gamemodes/nuclear/nuclear.dm
#	code/game/objects/items.dm
#	code/game/objects/objs.dm
#	code/game/objects/structures/grille.dm
#	code/modules/clothing/outfits/ert.dm
#	code/modules/clothing/suits/_suits.dm
#	code/modules/food_and_drinks/kitchen_machinery/griddle.dm
#	code/modules/jobs/job_types/head_of_security.dm
#	code/modules/jobs/job_types/shaft_miner.dm
#	code/modules/mob/living/carbon/human/inventory.dm
#	code/modules/recycling/disposal/bin.dm

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	code/modules/clothing/clothing.dm
#	code/modules/mob/living/carbon/human/species_types/plasmamen.dm

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	code/game/objects/items.dm

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	_maps/map_files/generic/CentCom.dmm
#	code/__DEFINES/uplink.dm
#	code/game/machinery/suit_storage_unit.dm
#	code/modules/clothing/outfits/ert.dm
#	code/modules/clothing/outfits/standard.dm

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	code/modules/clothing/clothing.dm

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	_maps/map_files/RadStation/RadStation.dmm
#	code/_globalvars/traits/_traits.dm
#	code/game/objects/items/chrono_eraser.dm
#	code/modules/mob/mob.dm

Tsar-Salat committed:
dejaku circuit fix

Tsar-Salat committed:
rad fix

Tsar-Salat committed:
make the constructor trait actually do something

Tsar-Salat committed:
no double

Tsar-Salat committed:
tra

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

PowerfulBacon reviewed (changes requested):
## Issues

Examining the infiltrator modsuit is broken. If you are holding it in your hand a few times, sometimes a giant wall of text appears (with some pretty poorly written lore containing grammatical issues) seperately as part of to_chat instead of inside of the examine blocks.

## General Comments and UX

This is a pretty big nerf to nuclear operatives due to a few things:
- There shouldn't be access requirements on any modsuits, including syndicate ones.
- The armour booster cannot be deployed while the suit is EVA capable.
- The suit now provides a large slowdown constantly, this is fine for ranged nukies but it means that there is no armour you can use if you want to go melee. The infiltrator suit doesn't protect you from space
- There are 2 buttons which don't really have a clearly seperate function. Deploy modsuit and activate modsuit should be integrated, if you fully deploy a modsuit then it should just automatically enable. If you undeploy any section of the suit then it should disable.
- Charge is now important to modsuits, however it is not shown without opening up the MOD interface panel. If charge is going to be a critical thing you need to know, there should be a UI icon somewhere (easiest way would be through a status effect with the duration text being the charge)
- The infiltrator suit is worn on the belt slot instead of the backpack slot, leading to some really strange interactions since you have to wear multiple suits. It gets extremely confusing and is really buggy, I managed to get it so that activating an elite hardsuit would undeploy the helmet to be active which broke the suit.

The UX for controlling modsuits is pretty poor. Particularly, with the syndicate control panel green means disabled and red means enabled which is extremely confusing:

![image](https://github.com/user-attachments/assets/cb7eeb41-6959-4744-a9b8-4274a381766b)

Note a huge fan of the new syndicate elite hardsuit sprite, most of the suit is fully black without any depth or texture but then there are some really weird looking orange pixels on it. The previous suit had a green flexible layer and black armour plating.

![image](https://github.com/user-attachments/assets/fcb425a3-57ac-4947-a105-9444c19d972d)

![image](https://github.com/user-attachments/assets/4152b1b6-6147-4cb5-87b7-1a0bdb99d26c)

And one minor thing, how is an entire suit of space worthy armour just suddenly appearing on my body simply because I am wearing a tiny backpack. It would have made far more sense as a core chestpiece than as a backpack.

They are cool all in all but feel super fiddly due to various points of poor user experience design. Improving the fiddliness of them should be (relatively) small changes which would have a big impact.

## Suggestions

- Completely remove the ability to activate sections (the clothing pieces) of the suit individually and remove the activate modsuit button. Change it so there is a single deploy/undeploy button which either deploys all parts of the suit and enables, or disables and undeploys.
- Make the infiltrator suit space proof. You have to spend additional TC to get it, so just make the advantage of it that its fast and space proof.
- Make the infiltrator suit worn on the backpack slot instead of the belt slot.
- Remove the deployability of the armour module, and instead make it have a slowdown. Reduce the base slowdown of armoured suits such as the syndicate suits to compensate. (Allowing them to be armoured even while in space)
- Change the syndicate modsuit panel so that green = enabled and red = disabled.
- Update the MOD interface panel so that the icons on the actions change depending on enabled/disabled; font-awesome likely supports a power off and unpin icon.
- Remove ID locking on modsuits.
- At the bare minimum, display the current power in the status menu. Preferably, make modsuits give you a status effect while enabled which show the current power as a percentage in place of the time remaining on that status effect.
- Make all default modules start pinned. Currently some modules get auto-pinned, while others (such as the flashlight) don't which is confusing. Pin everything by default, and let the user unpin them (if they didn't need the module they wouldn't install it).

I would also suggest making them work on the armour slot rather than the backpack slot, though I imagine that would be a pretty heft change.

Tsar-Salat committed:
moves slowdown per-part

https://github.com/tgstation/tgstation/pull/89632 https://github.com/tgstation/tgstation/pull/89657 https://github.com/tgstation/tgstation/pull/90116

Tsar-Salat committed:
swap syndicate UI colors

PowerfulBacon reviewed (dismissed):
I would prefer some form of solution to some of the problems I mentioned, but I will break down my request into a set of minimum requirements.

- Merge the deploy and activate modsuit button into a single button. I would recommend making it just automatically enable when it makes sense to be deployed (the chestplate is active), but that logic or how this is done is up to you.
- Find a way to make it less confusing when you are wearing 2 modsuits, or disable the ability to wear 2 modsuits by making them back slot only. You could do this by merging the buttons but it would be a lot more complex than making them mutually exclusive.
- Change the syndicate modsuit panel so that green = enabled and red = disabled.
- Make the power more visible to the user somewhere on the screen without the requirement for a TGUI window to be open.
- Show the complexity in the description of the exosuit fabricator recipes.
- When examining a MOD shell, give a hint that a MOD core needs to be obtained from cargo.

I am willing to find middleground on any of these issues, but these are what I would like to see at a minimum to make the MODsuit experience a little bit smoother.

If these requests seem unreasonable, we can have a discussion on discord to hear out the different sides and intended design of MODsuits within Bee with other maintainers who have used them.

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	_maps/map_files/BoxStation/BoxStation.dmm
#	_maps/map_files/generic/CentCom.dmm
#	beestation.dme
#	code/__DEFINES/antagonists.dm
#	code/__DEFINES/dcs/signals/signals_datum/signals_datum.dm
#	code/__DEFINES/exosuit_fab.dm
#	code/__DEFINES/is_helpers.dm
#	code/__DEFINES/mobs.dm
#	code/__DEFINES/obj_flags.dm
#	code/__DEFINES/traits/declarations.dm
#	code/__DEFINES/traits/sources.dm
#	code/__HELPERS/mobs.dm
#	code/_globalvars/traits/_traits.dm
#	code/datums/mind.dm
#	code/game/gamemodes/clown_ops/clown_ops.dm
#	code/game/gamemodes/nuclear/nuclear.dm
#	code/game/objects/items.dm
#	code/game/objects/items/defib.dm
#	code/game/objects/items/robot/robot_items.dm
#	code/game/objects/items/stacks/sheets/mineral/metals.dm
#	code/game/objects/items/stacks/stack.dm
#	code/game/objects/items/storage/boxes.dm
#	code/modules/antagonists/clock_cult/scriptures/interdiction_lens.dm
#	code/modules/antagonists/heretic/heretic_living_heart.dm
#	code/modules/antagonists/heretic/magic/eldritch_blind.dm
#	code/modules/antagonists/heretic/magic/eldritch_emplosion.dm
#	code/modules/antagonists/heretic/magic/eldritch_shapeshift.dm
#	code/modules/antagonists/role_preference/role_roundstarts.dm
#	code/modules/clothing/outfits/ert.dm
#	code/modules/clothing/outfits/standard.dm
#	code/modules/jobs/job_types/cargo_technician.dm
#	code/modules/mob/living/carbon/human/human_movement.dm
#	code/modules/mob/living/carbon/human/species.dm
#	code/modules/mob/living/carbon/human/species_types/diona.dm
#	code/modules/mob/living/carbon/human/species_types/ethereal.dm
#	code/modules/mob/living/carbon/human/species_types/jellypeople.dm
#	code/modules/mob/living/carbon/human/species_types/lizardpeople.dm
#	code/modules/mob/living/carbon/human/species_types/mothmen.dm
#	code/modules/mob/living/carbon/human/species_types/oozelings.dm
#	code/modules/mob/living/carbon/human/species_types/snail.dm
#	code/modules/mob/living/silicon/robot/robot.dm
#	code/modules/mob/living/silicon/robot/robot_modules.dm
#	code/modules/mob/mob.dm
#	code/modules/research/xenobiology/crossbreeding/_status_effects.dm
#	code/modules/research/xenobiology/crossbreeding/stabilized.dm
#	code/modules/surgery/organs/heart.dm
#	code/modules/surgery/surgery.dm
#	code/modules/unit_tests/_unit_tests.dm

Tsar-Salat committed:
fixes

Tsar-Salat committed:
.

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

# Conflicts:
#	code/datums/components/gps.dm
#	code/modules/clothing/spacesuits/hardsuit.dm
#	tgui/packages/tgui/styles/themes/syndicate.scss

Tsar-Salat committed:
fix AI runtime

Tsar-Salat committed:
fix stabilized extracts

Tsar-Salat committed:
moves ninja entirely to modsuits

Tsar-Salat committed:
fix

Tsar-Salat committed:
cant sleep

Tsar-Salat committed:
fix shuttle issue

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

Tsar-Salat committed:
autowiki test

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into modsuits-take5

Tsar-Salat committed:
energy net fix

Dejaku51 reviewed (approved):
There aren't any issues known anymore, code looks good as far as im aware. Its good to merge in my opinion
